ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixParticipantTest.java

```diff
@@ -63,19 +65,97 @@
     helixManagerFactory = new MockHelixManagerFactory();
   }
 
+  private ReplicaId createMockReplicaId(String partitionIdString) {
+    ReplicaId replicaId = Mockito.mock(ReplicaId.class);
+    PartitionId partitionId = Mockito.mock(PartitionId.class);
+    Mockito.when(partitionId.toPathString()).thenReturn(partitionIdString);
+    Mockito.when(replicaId.getPartitionId()).thenReturn(partitionId);
+    return replicaId;
+  }
+
+  private void listIsExpectedSize(List list, int expectedSize, String name) {
+    assertNotNull(name + " is null", list);
+    assertEquals(name + " is not size " + expectedSize, expectedSize, list.size());
+  }
+
+  /**
+   * Tests setReplicaSealedState method for {@link HelixParticipant}
+   * @throws IOException
+   */
+  @Test
+  public void testSetReplicaSealedState() throws IOException {
+    //setup HelixParticipant and dependencies
+    String partitionId = "somePartitionId";
+    String partitionId2 = "someOtherPartitionId";
+    ReplicaId replicaId = createMockReplicaId(partitionId);
+    ReplicaId replicaId2 = createMockReplicaId(partitionId2);
+    String hostname = "localhost";
+    int port = 2200;
+    String instanceName = ClusterMapUtils.getInstanceName(hostname, port);
+    HelixParticipant helixParticipant =
+        new HelixParticipant(new ClusterMapConfig(new VerifiableProperties(props)), helixManagerFactory);
+    helixParticipant.initialize(hostname, port, Collections.EMPTY_LIST);
+    HelixManager helixManager = helixManagerFactory.getZKHelixManager(null, null, null, null);
+    HelixAdmin helixAdmin = helixManager.getClusterManagmentTool();
+    InstanceConfig instanceConfig = new InstanceConfig("someInstanceId");
+    helixAdmin.setInstanceConfig(clusterName, instanceName, instanceConfig);
+    instanceConfig = helixAdmin.getInstanceConfig(clusterName, instanceName);
+    InstanceConfig ic = helixAdmin.getInstanceConfig(clusterName, instanceName);
+
+    //Make sure the current sealedReplicas list is null
+    List<String> sealedReplicas = ClusterMapUtils.getSealedReplicas(instanceConfig);
+    assertNull("sealedReplicas is not null", sealedReplicas);
+
+    String listName = "sealedReplicas";
+
+    //Check that invoking setReplicaSealedState adds the partition to the list of sealed replicas
+    helixParticipant.setReplicaSealedState(replicaId, true);
+    sealedReplicas = ClusterMapUtils.getSealedReplicas(helixAdmin.getInstanceConfig(clusterName, instanceName));
+    listIsExpectedSize(sealedReplicas, 1, listName);
+    assertEquals(partitionId, sealedReplicas.get(0));
+
+    //Seal another replicaId
+    helixParticipant.setReplicaSealedState(replicaId2, true);
+    sealedReplicas = ClusterMapUtils.getSealedReplicas(helixAdmin.getInstanceConfig(clusterName, instanceName));
+    listIsExpectedSize(sealedReplicas, 2, listName);
+    assertEquals(partitionId2, sealedReplicas.get(1));
+
+    //Check that sealed replica list doesn't take duplicates
```

Comment:
when you do this test,  perhaps you should use a different replicaId instance but with the same partition - just to ensure that the prod code is not doing an equality check on replicaId somewhere (which it is not).